src/ui/views/native/AddBookmarkIntent.vue

```diff
@@ -192,6 +186,16 @@ export default {
     }
   }
 }
+
+function checkUrl(url) {
+  try {
+    // eslint-disable-next-line
+    new URL(url)
+    return null
+  } catch (e) {
+    return 'Invalid URL'
+  }
+}
```

Comment:
If you put this function in the `methods` section above with the other methods, you can call it as `this.checkUrl` from the url watcher and the data provider function ;)

Reply:
Thanks, I have done that, and rebased on the latest develop. Not sure if you normally rebase, but since this is such a small change I thought it would probably be OK?